### DISCUSSION TITLE
Deploying to server

### DISCUSSION THREAD

**Discussion by proton029** [Budibase/budibase#7053] [Help]
Hi. I have built an app in budibase and published it on local system running docker desktop and now i want to deploy it to a server. What are the steps to do that?.

**Comment by jonnymccullagh** [comment]
Hi @proton029, just checking whether you are asking how to migrate the data from a local container to a server container? Or if you are asking how to setup budibase on a server? 
Have you already got a server with a provider (such as AWS, DigitalOcean, Linode) ?

If you already have a server running somewhere with docker you could move the container from your local machine to that server using the first tip suggested [here](https://bobcares.com/blog/move-docker-container-to-another-host/).

**Reply by proton029** [reply]
Hi thanks for replying. I wanted to verify whether i have to move the container which have budibase builder whom I have used to build the app to the server and one more thing is that how can i change the local host part of the app link in builder.?

**Reply by jonnymccullagh** [reply]
The data for the app you have built is stored in a CouchDB database inside the budibase container on your local desktop. You can export this data but first you need somewhere to import it to.
You could consider using the [Budibase Cloud](https://account.budibase.app/register) or [self-hosting](https://docs.budibase.com/docs/hosting-methods) on a server. Using either of these methods should get you an internet-accessible web address rather than 'localhost'. 

When you have that step done you could consider exporting your app from your desktop and importing it to your server installation:
[Export/Import Doc](https://docs.budibase.com/docs/portal#export-app )

**Reply by proton029** [reply]
budibase cloud doesn't helping out because its keep logging me out once i have imported my app. then after cant login
